fix: replace abandoned php-http/message-factory with psr/http-factory

kochen opened the pull request:
fixes #164

wazelin reviewed (changes requested):
Thanks for this contribution, @kochen!
My apologies for not reacting to it earlier. I had the whole November very far away from my laptop.

I feel like we should drop any HTTP factory usage, and just instantiate a `Nyholm\Psr7\Response` object explicitly. Places like [this one](https://github.com/oat-sa/lib-lti1p3-core/blob/6.9.0/src/Security/Oidc/Server/OidcInitiationRequestHandler.php#L64) require headers that cannot be passed through `psr/http-factory`.

I'm not too concerned with the change being a breaking one, as we're going to drop support for PHP 7.4 anyway.

In summary: let's get the responses fixed. Then this and the other PRs look good-to-go.

kochen commented:
@wazelin the change that I proposes does not require any additional changes (and passes all tests).


I suggest we firstdeal with the reported vulnerability and then look at the rest.
Could you approve/merge this #169 first?

kochen commented:
@wazelin I refactored the code to use `Nyholm\Psr7\Response` (and removed the abandoned `php-http/message-factory`).

There are 2 commits here, because I'm using the "bumped" versions from #166

kochen commented:
Sure, though I would still suggest tackling these tasks independently and also merging them one by one.

kochen commented:
@wazelin the libraries cannot be updates, because they rely on core v6.9 which uses the ResponseFactory.

Could you maybe release v7-beta/RC so the code adjustments could be done?

wazelin commented:
> @wazelin the libraries cannot be updates, because they rely on core v6.9 which uses the ResponseFactory.
> 
> Could you maybe release v7-beta/RC so the code adjustments could be done?

I'm not quite sure I understand. Removing `ResponseFactory` usage doesn't really _require_ having that same factory removed from the core library, does it?
I mean, to me it seems that places [like this one here](https://github.com/oat-sa/lib-lti1p3-ags/blob/main/src/Service/Result/Server/Handler/ResultServiceServerRequestHandler.php#L135) should be refactored to return an instance of `Nyholm\Psr7\Response` explicitly.

kochen commented:
But places [like this](https://github.com/oat-sa/lib-lti1p3-ags/blob/main/tests/Integration/Service/LineItem/Server/Handler/CreateLineItemServiceServerRequestHandlerTest.php#L82) (and in many other tests), still expect the factory being passed.

wazelin commented:
I've opened alternative PRs for all of the lti1p3 libraries and the Symfony bundle, @kochen.
Let me know what you think.

kochen commented:
looks great!

I'm following... ;)
